hudi-common/src/main/java/org/apache/hudi/common/util/ObjectSizeCalculator.java

```diff
@@ -92,6 +92,7 @@ public static long getObjectSize(Object obj) throws UnsupportedOperationExceptio
   private final Set<Object> alreadyVisited = Collections.newSetFromMap(new IdentityHashMap<>());
   private final Deque<Object> pending = new ArrayDeque<>(64);
   private long size;
+  private static final ThreadLocal<ObjectSizeCalculator> objectSizeCalculator = ThreadLocal.withInitial(() -> new ObjectSizeCalculator(CurrentLayout.SPEC));
```

Comment:
Let's bubble this up to the top of the declarations list (where all the static members are)

Reply:
+1